napari_cellseg3d/code_models/worker_training.py

```diff
@@ -0,0 +1,786 @@
+                        * self.config.validation_percent
+                    )
+                    num_val_samples = ceil(
+                        self.config.num_samples
+                        * (1 - self.config.validation_percent)
+                    )
+                    sample_loader_train = get_loader_func(num_train_samples)
+                    sample_loader_eval = get_loader_func(num_val_samples)
+                else:
+                    num_train_samples = (
+                        num_val_samples
+                    ) = self.config.num_samples
+
+                    sample_loader_train = get_loader_func(num_train_samples)
+                    sample_loader_eval = get_loader_func(num_val_samples)
+
+                logger.debug(f"AMOUNT of train samples : {num_train_samples}")
+                logger.debug(
+                    f"AMOUNT of validation samples : {num_val_samples}"
+                )
+
+                logger.debug("train_ds")
+                train_ds = PatchDataset(
+                    data=self.train_files,
+                    transform=train_transforms,
+                    patch_func=sample_loader_train,
+                    samples_per_image=num_train_samples,
+                )
+                logger.debug("val_ds")
+                val_ds = PatchDataset(
+                    data=self.val_files,
+                    transform=val_transforms,
+                    patch_func=sample_loader_eval,
+                    samples_per_image=num_val_samples,
+                )
+
+            else:
+                load_whole_images = Compose(
+                    [
+                        LoadImaged(
+                            keys=["image", "label"],
+                            # image_only=True,
+                            # reader=WSIReader(backend="tifffile")
+                        ),
+                        EnsureChannelFirstd(keys=["image", "label"]),
+                        Orientationd(keys=["image", "label"], axcodes="PLI"),
+                        QuantileNormalizationd(keys=["image"]),
+                        SpatialPadd(
+                            keys=["image", "label"],
+                            spatial_size=PADDING,
+                        ),
+                        EnsureTyped(keys=["image", "label"]),
+                    ]
+                )
+                logger.debug("Cache dataset : train")
+                train_ds = CacheDataset(
+                    data=self.train_files,
+                    transform=Compose(load_whole_images, train_transforms),
+                )
+                logger.debug("Cache dataset : val")
+                val_ds = CacheDataset(
+                    data=self.val_files, transform=load_whole_images
+                )
+            logger.debug("Dataloader")
+            train_loader = DataLoader(
+                train_ds,
+                batch_size=self.config.batch_size,
+                shuffle=True,
+                num_workers=2,
```

Comment:
Is 2 set as the default for the num_workers? Should default be no multithreading?

Reply:
Changed to self.config.num_workers; this defaults to 4. Do you know what issues might arise should 4 not be available ? Does it default back to None, or raise an error ?